src/Features/CSharp/Portable/Structure/Providers/RegionDirectiveStructureProvider.cs

```diff
@@ -55,7 +55,7 @@ private static string GetBannerText(DirectiveTriviaSyntax simpleDirective)
                     type: BlockTypes.PreprocessorRegion,
                     bannerText: GetBannerText(regionDirective),
                     autoCollapse: autoCollapse,
-                    isDefaultCollapsed: !options.IsMetadataAsSource));
+                    isDefaultCollapsed: options.CollapseRegionsOnFileOpen));
```

Comment:
This is where the "Enter outlining mode" was equivalent to "Collapse regions on file open", but has been split to a separate option. The `!options.IsMetadataAsSource` wasn't doing anything here (it may as well have been `true`) because the code path for metadata as source uses the `autoCollapse` property which has the real logic (a few lines up from what GitHub shows here)